turbo/transactions/tracing.go

```diff
@@ -65,7 +65,12 @@ func ComputeTxEnv(ctx context.Context, block *types.Block, cfg *params.ChainConf
 		}
 		// Ensure any modifications are committed to the state
 		// Only delete empty objects if EIP158/161 (a.k.a Spurious Dragon) is in effect
-		_ = statedb.FinalizeTx(vmenv.ChainRules(), state.NewNoopWriter())
+		_ = statedb.FinalizeTx(vmenv.ChainRules, reader)
+
+		if idx+1 == len(block.Transactions()) {
+			// Return the state from evaluating all txs in the block, note no msg or TxContext in this case
+			return nil, BlockContext, vm.TxContext{}, statedb, reader, nil
+		}
```

Comment:
This line doing same? https://github.com/ledgerwatch/erigon/pull/3148/files#diff-664a6f9734f5a173f4805bd1eee6763dab47fff1aec9839fdbef67036d7b09faR58

Reply:
sorry I did not quite understand this comment.

Reply:
nevermind